libs/domain/site/src/services/link/default-link.service.ts

```diff
@@ -43,6 +43,24 @@ export class DefaultLinkService implements LinkService {
     );
   }
 
+  isCurrent(url: string, exactMatch?: boolean): Observable<boolean> {
+    return this.routerService.currentRoute().pipe(
+      switchMap((currentRoute) => {
+        if (!currentRoute || !isRouterPath({ path: currentRoute })) {
+          return throwError(() => new Error('Current route is not available'));
+        }
+
+        const currentUrl = `${this.baseRoute}${currentRoute}`;
+
+        if (exactMatch) {
+          return of(currentUrl === url);
+        } else {
+          return of(currentUrl.startsWith(url));
```

Comment:
there may be some edge cases, where `startsWith` may not always work as expected

Reply:
what edge cases?